lib/App.js

```diff
@@ -109,22 +133,26 @@ class App {
 
     let promise    = Q();
     const response = new Response();
-    _.each(this._middleware, (route) => {
-      promise = promise.then(() => {
-        if (!response.ended) {
-          return route.execute({ request, response });
-        }
-      });
-    });
-    _.each(this._errorMiddleware, (route) => {
-      if (!route.isMatch(request)) {
-        return;
+    _.each(this._middleware, (registeredRoute) => {
```

Comment:
Rename var `registeredRoute` to `middleware`. Cause you know...I'm iterating over middleware.